### PR TITLE
Throw exception on non-existent key for `update-key`

### PR DESCRIPTION
**Pull request by devleifr**

### CODE REVIEW

File: src/darkleaf/di/core.clj [hunk 1]
```diff
@@ -507,27 +507,30 @@
...
+                             (let [t    (deps new-key)
+                                   f    (deps f-key)
+                                   args (map deps arg-keys)]
+                               (apply f t args)))
+                           (demolish [_ _]))
+          own-registry   (zipmap (cons f-key arg-keys)
+                                 (cons f     args))
+          target-factory (registry target)]
+      (when (nil? target-factory)
+        (throw (NoSuchElementException. (format "Can't update non-existent key %s" target))))
```
Comment (darkleaf) [inline src/darkleaf/di/core.clj:529]: `(str "msg " target)`

File: src/darkleaf/di/core.clj [hunk 2]
```diff
@@ -507,27 +507,30 @@
...
+                             (let [t    (deps new-key)
+                                   f    (deps f-key)
+                                   args (map deps arg-keys)]
+                               (apply f t args)))
+                           (demolish [_ _]))
+          own-registry   (zipmap (cons f-key arg-keys)
+                                 (cons f     args))
+          target-factory (registry target)]
+      (when (nil? target-factory)
+        (throw (NoSuchElementException. (str "Can't update non-existent key " target))))
```
Comment (darkleaf) [inline src/darkleaf/di/core.clj:529]: @KGOH how do you think is it a breaking change or a bug fix?

Comment (KGOH) [reply src/darkleaf/di/core.clj:529]: let's publish this as a bug fix; if someone complains, we'll think what to do

Comment (KGOH) [reply src/darkleaf/di/core.clj:529]: ex-info :type ::missing-dependency :message "Can't update...

**Review by KGOH — CHANGES_REQUESTED** [review]

**Pull request merged**